sycl/test-e2e/Config/select_device.cpp

```diff
@@ -167,285 +167,215 @@ static std::vector<DevDescT> getAllowListDesc(std::string allowList) {
   return decDescs;
 }
 
-int main() {
-  bool passed = false;
-  std::string sycl_be("opencl");
-  if (getenv("ONEAPI_DEVICE_SELECTOR")) {
-    sycl_be = getenv("ONEAPI_DEVICE_SELECTOR");
+bool is_known_be(backend be) {
+    backend known_bes[] = {backend::opencl, backend::ext_oneapi_level_zero,
```

Comment:
do we have to worry about esmid emulator backend?

Reply:
No entry - no problems :D

That part is NFC and I tried to follow what the tests did before my change (as long as it wasn't an obvious bug/mistake).